JASP-Desktop/widgets/boundqmlcomponentslist.cpp

```diff
@@ -133,27 +134,46 @@ Option* BoundQMLComponentsList::createOption()
 	}
 	else
 	{
-		QVariant defaultValuesVar = getItemProperty("defaultValues");
-		QList<QVariant> defaultValues = defaultValuesVar.toList();
+		QString			defaultName			= getItemProperty("newItemName").toString();
+		QVariant		defaultValuesVar	= getItemProperty("defaultValues");
+		QList<QVariant> defaultValues		= defaultValuesVar.toList();
+		int				minimumItems		= getItemProperty("minimumItems").toInt();
+
+		while (defaultValues.length() < minimumItems)
+			defaultValues.push_back(defaultName);
+
+		QList<QString> keyValues;
+		QString keyName = tq(_optionKeyName);
 
 		for (const QVariant& defaultValue : defaultValues)
 		{
+			QVariantMap defaultValueMap = defaultValue.toMap();
 			Options* row = dynamic_cast<Options*>(templote->clone());
 			OptionVariables* optionVar = new OptionVariable();
-			optionVar->setValue(_incrementCounter().toStdString());
+
+			QString keyValue = (defaultValue.type() == QVariant::String) ? defaultValue.toString() : defaultName;
+			if (defaultValueMap.contains(keyName))
+				keyValue = defaultValueMap[keyName].toString();
+
+			keyValue = _makeUnique(keyValue, keyValues);
+			keyValues.push_back(keyValue);
+
+			optionVar->setValue(keyValue.toStdString());
 			row->add(_optionKeyName, optionVar);
 
-			QVariantMap defaultValueMap = defaultValue.toMap();
 			QMapIterator<QString, QVariant> it(defaultValueMap);
 			while (it.hasNext())
 			{
 				it.next();
 				QString name = it.key();
-				QVariant value = it.value();
 
-				Option* option = row->get(name.toStdString());
-				if (option)		option->set(value.toJsonValue().toString().toStdString());
-				else			Log::log() << "Default value with name " << name << " is unknown in the component" << std::endl;
+				if (name != keyName)
+				{
+					QVariant value = it.value();
+					Option* option = row->get(name.toStdString());
+					if (option)		option->set(value.toJsonValue().toString().toStdString());
+					else			Log::log() << "Default value with name " << name << " is unknown in the component" << std::endl;
```

Comment:
I wouldn't know what this errormsg means... Not important enough to block this PR but, maybe it could be clarified a bit?